src/components/chip.tsx

```diff
@@ -68,25 +70,34 @@ export const Chip: FC<ChipProps & HTMLAttributes<HTMLButtonElement>> = ({
     },
     []
   );
+  const props = { ...rest };
+  let element: 'button' | 'span' = 'button';
+  if (onRemove) {
+    element = 'span';
+    if (onClick || onKeyPress) {
+      props.role = 'button';
+      props.tabIndex = 0;
+    }
+  }
 
+  const Element = element;
   return (
-    <span
-      role="button"
+    <Element
       className={cn(
         'chip',
         { 'chip--disabled': disabled, 'chip--compact': compact },
         className
       )}
+      type={element === 'button' ? 'button' : undefined}
```

Comment:
Couldn't figure out a way to assign to props instead of inline as kept getting a type error.

Reply:
I guess that's fine like this, typing an extra props object with part of the props, but not all of them, and changing conditionally, also proved complicated for me to before.